filters/imagefilter.go

```diff
@@ -80,8 +80,17 @@ func errorResponse() *http.Response {
 
 func buildParameters(ctx filters.FilterContext, image *bimg.Image) *ImageFilterContext {
 	parameters := map[string][]string(nil)
+
 	if ctx != nil {
 		parameters = ctx.Request().URL.Query()
+
+		focalPointX := ctx.PathParam("focalPointX");
+		focalPointY := ctx.PathParam("focalPointY");
+
+		if len(focalPointX) > 0 && len(focalPointY) > 0 {
```

Comment:
I would not add the `PathParams` to the parameters, I would rather add all of the path params to the `ImageFilterContext`, something like this:

```
&ImageFilterContext{
 		Image:      image,
.....
                PathParams: pathParams
}
```

The reason is that with your approach, every time a new filter adds a new parameter we need to modify this method, which breaks the [Open Closed Principle](https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle). As soon as the path parameters are part of the ImageFilterContext, each of the filters can access them.

Reply:
@danpersa the problem is that Skipper currently does not allow to get all PathParams. You need to access them by name.

I can move them out of the parameters but unless there is a change in Skipper I don't see a way to add all PathParams without knowing which exist.